ethicml/algorithms/inprocess/agarwal_reductions.py

```diff
@@ -62,24 +63,20 @@ def __init__(
             kernel: Kernel type for the SVM algorithm.
             seed: Random seed.
         """
-        if fairness not in VALID_FAIRNESS:
-            raise ValueError(f"results: fairness must be one of {VALID_FAIRNESS!r}.")
-        if classifier not in VALID_MODELS:
-            raise ValueError(f"results: classifier must be one of {VALID_MODELS!r}.")
         self.seed = seed
         self.model_dir = dir if isinstance(dir, Path) else Path(dir)
         chosen_c, chosen_kernel = settings_for_svm_lr(classifier, C, kernel)
         self.flags: AgarwalArgs = {
-            "classifier": classifier,
-            "fairness": fairness,
+            "classifier": str(classifier),
+            "fairness": str(fairness),
```

Comment:
I think this should be:

```suggestion
            "classifier": classifier.name,
            "fairness": fairness.name,
```
because `str()` of an enum is `"EnumClass.name"`.

Reply:
I'm using ranzen's `@enum_as_str` decorator. Maybe it would be clearer to just use `.name`

Reply:
Ah, sorry I didn't see that. No, everything is fine then.